test/integration/hotrestart_test.sh

```diff
@@ -153,4 +153,23 @@ do
   TEST_INDEX=$((TEST_INDEX+1))
 done
 
+# set -e forces the script to exit on non-zero exit codes. Set +e makes it easier to
```

Comment:
Instead of copying this, can we factor this and the postfix into bash functions like `saveHeapCheck` and `restoreHeapCheck` ?